### PR TITLE
Add top level refund methods.

### PR DESCRIPTION
**Pull request by cupcait**
r? @kyleconroy

### CODE REVIEW

File: tests/RefundTest.php [hunk 1]
```diff
@@ -8,28 +8,42 @@ class RefundTest extends TestCase
...
+    public function testListForCharge()
     {
         $charge = self::createTestCharge();
-        $refA = $charge->refunds->create(array('amount' => 50));
-        $refB = $charge->refunds->create(array('amount' => 50));
+        $refA = Refund::create(array('amount' => 100, 'charge' => $charge->id));
+        $refB = Refund::create(array('amount' => 50, 'charge' => $charge->id));
 
-        $all = $charge->refunds->all();
+        $all = Refund::all('charge' => $c);
```
Comment (bkrausz) [inline tests/RefundTest.php]: PHP, in all its verbose horrors, needs `array('charge' => $c)`

File: tests/RefundTest.php [hunk 2]
```diff
@@ -8,28 +8,42 @@ class RefundTest extends TestCase
     public function testCreate()
     {
         $charge = self::createTestCharge();
-        $ref = $charge->refunds->create(array('amount' => 100));
-        $this->assertSame(100, $ref->amount);
-        $this->assertSame($charge->id, $ref->charge);
+        $refund = Refund::create(array('amount' => 100, 'charge' => $charge->id));
```
Comment (bkrausz) [inline tests/RefundTest.php]: Can we leave a few tests in place that the old refund creation call style still works?

**Comment by bkrausz** [comment]
Some style errors:
```
FILE: /home/travis/build/stripe/stripe-php/tests/RefundTest.php
----------------------------------------------------------------------
FOUND 8 ERRORS AFFECTING 7 LINES
----------------------------------------------------------------------
 40 | ERROR | [x] Opening brace indented incorrectly; expected 5
    |       |     spaces, found 4
 51 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     4
 51 | ERROR | [x] Closing brace indented incorrectly; expected 5
    |       |     spaces, found 4
 53 | ERROR | [x] Line indented incorrectly; expected at least 8
    |       |     spaces, found 4
 54 | ERROR | [x] Line indented incorrectly; expected at least 8
    |       |     spaces, found 4
 55 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
 57 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
 59 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
----------------------------------------------------------------------
PHPCBF CAN FIX THE 8 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Comment by cupcait** [comment]
Yup, thanks! I forgot to put a note that this is a WIP (currently debugging). Will address the comments and ping here when tests pass and I think it's ready for another look.

**Comment by bkrausz** [comment]
Gah sorry, I forgot I get CCed on all PRs, I assumed there was an `r? @bkrausz` there somewhere. Nevermind :)

**Comment by cupcait** [comment]
No worries :) I probably would have needed some of this feedback eventually anyway!

**Comment by cupcait** [comment]
r? @bkrausz Ready for another look!
cc @kyleconroy 

I added the old tests back in (I can remove some if you think it's overkill). I'm also a bit unsure about my new testList(), which uses existing data on a heavily used test account. I think this should be pretty stable but am open to feedback.

**Comment by bkrausz** [comment]
lgtm!

**Pull request merged**